internal/parser/parser/email_parser_test.go

```
@@ -464,3 +781,216 @@ func TestEmailParser_AttachmentsWithoutBody(t *testing.T) {
+// collection stores the byte-exact decoded-CTE bytes under "raw" alongside the
+// charset-decoded "payload". Declaring charset=gbk makes decodeMailPayload
+// decode GBK bytes to a UTF-8 string, so the two differ — re-chunk relies on
+// "raw" to avoid silently corrupting the attachment.
+func TestReadMailBody_AttachmentPreservesRaw(t *testing.T) {
+	// GBK-encoded "中文" (中=0xD6D0, 文=0xCEC4).
+	gbk := []byte{0xD6, 0xD0, 0xCE, 0xC4}
+
+	var buf bytes.Buffer
+	mw := multipart.NewWriter(&buf)
+	bp, _ := mw.CreatePart(textproto.MIMEHeader{"Content-Type": {"text/plain"}})
+	bp.Write([]byte("body"))
```

golangci-lint (2.12.2)

[error] Error return value of `bp.Write` is not checked

(errcheck)